c++/src/kj/filesystem-disk-unix.c++

```diff
@@ -383,7 +383,12 @@ public:
   }
 
   void zero(uint64_t offset, uint64_t size) const {
-#ifdef FALLOC_FL_PUNCH_HOLE
+// If FALLOC_FL_PUNCH_HOLE is defined, use it to efficiently zero the area.
+//
+// A fallocate() wrapper was only added to Android's Bionic C library as of API level 21,
+// but FALLOC_FL_PUNCH_HOLE is apparently defined in the headers before that, so we'll
+// have to explicitly test for that case.
+#if defined(FALLOC_FL_PUNCH_HOLE) && !(__ANDROID__ && __BIONIC__ && __ANDROID_API__ < 21)
```

Comment:
Sorry, only realized after seeing this applied that I mis-indented the comments.

```suggestion
    // If FALLOC_FL_PUNCH_HOLE is defined, use it to efficiently zero the area.
    //
    // A fallocate() wrapper was only added to Android's Bionic C library as of API level 21,
    // but FALLOC_FL_PUNCH_HOLE is apparently defined in the headers before that, so we'll
    // have to explicitly test for that case.
#if defined(FALLOC_FL_PUNCH_HOLE) && !(__ANDROID__ && __BIONIC__ && __ANDROID_API__ < 21)
```